# NRPminer 0.2: Scalable Metabologenomics Algorithm for Non-Ribosomal Peptide Discovery 

## About NRPminer

NRPminer is the first scalable and modification-tolerant bioinformatics tool for NRP discovery by integrating large (meta)genomic and tandem mass spectrometry (metabolomics) datasets. NRPminer uses efficient algorithm to Non-Ribosomal Peptides (NRPs) represent a biomedically important class of natural products that include a multitude of antibiotics and other clinically used drugs. NRPs are not directly encoded in the genome but are instead produced by metabolic pathways encoded by biosynthetic gene clusters (BGCs). Since the existing genome mining tools predict many putative NRPs synthesized by a given BGC, it remains unclear which of these putative NRPs are correct and how to identify post-assembly modifications of amino acids in these NRPs in a blind mode, without knowing which modifications exist in the sample. 

To address this challenge, we developed NRPminer, a modification-tolerant tool for NRP discovery from large (meta)genomic and mass spectrometry datasets. Our results shows that NRPminer is able to identify many NRPs from different environments, including completly NRP families. 

## NRPminer pipeline


![alt text](docs/nrpminer_pipeline.jpg "NRPminer pipeline")

NRPminer pipeline. NRPminer steps invovles (a) Predicting NRPS BGCs using antiSMASH16. Each ORF is represented by an arrow, and each A-domain is represented by a square, (b) predicting putative amino acids for each NRP residue using NRPSpredictor215, colored circles represents different amino acids (AAs), (c) generating multiple assembly lines by considering various combinations of ORFs and generating all putative core NRPs for each assembly line in the identified BGC (for brevity only assembly lines generated by deleting a single NRPS unit are shown; in practice, NRPminer considers loss of up to two NRPS units, as well as single and double duplication of each NRPS unit), (d) filtering the core NRPs based on their specificity scores, (e) identifying domains corresponding to known modifications and incorporating them in the selected core NRPs (modified amino acids are represented by purple squares), (f) generating linear, cyclic and branch-cyclic backbone structures for each core NRP, (g) generating a set of high-scoring PSMs using modification-tolerant search of spectra against the database of the constructed putative NRP structures. NRPminer considers all possible mature NRPs with up to one PAM (shown as hexagons) in each NRP structure. For brevity some of the structures are not shown. (h) computing statistical significance of PSMs and reporting the significant PSMs, and (i) expanding the set of identified spectra using spectral networks58. Nodes in the spectral network represent spectra and edges connect “similar” spectra.


## Dependencies (add binaries to PATH):

	- NPDtools (https://github.com/ablab/npdtools)

## System requirements:

	- Linux or macOS
	- Python 2.7
	- Biopython 
	- GNU sed 


## To install and run test:

     make Makefile


If the test is finished correctly, you will see the identification of Longicatemycin. 

## Usage examples: 


	nrpminer.py -s [spectrum_file]  --antismash_resgbk [path_to_gbk_file_generated_by_antiSMASH] --orfDel [num_orfs_deleted] --derepdir [path_to_NPDtools_bin] --maxmod [maximum_modification_mass] -o [output_folder] --pval_threshold [pvalue_threshold]      


For the full list of available options please run

     nrpminer.py -h


Output:

* nrpminer_significant_psms.txt:                    list of all PSMs with significant PSM P-value as well as the sequence of the identified NRP


